application/classes/Ushahidi/Listener/IncomingMessageListener.php

```diff
@@ -0,0 +1,78 @@
+<?php defined('SYSPATH') or die('No direct script access');
+
+/**
+ * Ushahidi Incoming Message Listener
+ *
+ * Listens for incoming messages,
+ *
+ * @author     Ushahidi Team <team@ushahidi.com>
+ * @package    Ushahidi\Application
+ * @copyright  2018 Ushahidi
+ * @license    https://www.gnu.org/licenses/agpl-3.0.html GNU Affero General Public License Version 3 (AGPL3)
+ */
+
+use League\Event\AbstractListener;
+use League\Event\EventInterface;
+use Ushahidi\Core\Entity\FormAttributeRepository;
+use Ushahidi\Core\Entity\TargetedSurveyStateRepository;
+use Ushahidi\Core\Entity\MessageRepository;
+
+class Ushahidi_Listener_IncomingMessageListener extends AbstractListener
+{
+    public function setFormAttributeRepo(FormAttributeRepository $form_attr_repo)
+	{
+		$this->form_attr_repo = $form_attr_repo;
+	}
+
+    public function setTargetedSurveyStateRepo(TargetedSurveyStateRepository $tss_repo)
+    {
+        $this->tss_repo = $tss_repo;
+    }
+
+    public function handle(EventInterface $event, $event_data = [])
+    {
+        /* @TODO: determing how should we mark a survey as done —
```

Comment:
shouldn't we have a single record per contact+form in targeted survey state?

Reply:
last_sent_form_attribute_id I assumed meant that null=not sent. Because If it's finished, shouldn't this be the last question sent?

Reply:
we might want to merge contact_post_state and this, btw